drivers/can/can_sam.c

```diff
@@ -101,7 +102,12 @@ static int can_sam_init(const struct device *dev)
 		return ret;
 	}
 
-	ret = can_mcan_configure_mram(dev, 0U, sam_cfg->mram);
+	/* use higher 16bit only, mcan sets lower 16bit */
+	uint32_t mrba = (sam_cfg->mram & 0xFFFF0000);
+	/* keep lower 16bit, update higher 16bit */
+	*sam_cfg->dma_base_reg = mrba | (*sam_cfg->dma_base_reg & 0x0000FFFF);
```

Comment:
I would read out the current value first, modify it, and write it back. The construct is here is not easily parseable.

Reply:
done